### ISSUE TITLE
SVC là như thế nào? *nên chú ý đặt tên biến cho hợp lý*

### ISSUE THREAD

**Issue by hoanguyentat** [tideburning/python_test4#2]
https://github.com/tideburning/python_test4/blob/9513b83bfbd4d258ee51fe40eaab853f764c1f30/bai1/bai1.py#L108-L113

**Comment by tideburning** [comment]
cái SVC là hàm mặc định trong thư viện của sklearn.svm ạ (được import ở dòng 14) ! cái chỗ này vì chạy ở máy em thấy nó nhanh hơn cái kia nên em đổi thuật toán thôi ạ(lười gõ lại code nên em để cái tên là knn - KNeighborClassifier), lần sau em sẽ chú ý hơn ạ !!!!

**Comment by hoanguyentat** [comment]
e mô tả lại ý tưởng của thuật toán svc là ntn ấy?

**Comment by tideburning** [comment]
Dạ, SVC là viết tắt của Support Vector Classification(Hỗ Trợ Phân Loại vector) hàm này sử dụng một ma trận Gram. Về thuật toán nó giống thằng SVM nhưng vì chọn được tham số hợp lý (trong trường hợp này) (một cách chính xác hơn là SVC là 1 ) nên nó chạy tốt hơn. 
Tham số mặc định của SVC như sau
SVC(C=1.0, kernel=’rbf’, degree=3, gamma=’auto’, coef0=0.0, shrinking=True, probability=False, tol=0.001, cache_size=200, class_weight=None, verbose=False, max_iter=-1, decision_function_shape=’ovr’, random_state=None)

C- Thông số phạt của giới hạn lỗi (float)

Kernel - Chỉ định loại hạt nhân được sử dụng trong thuật toán. Nó phải là một trong số ‘tuyến tính’, ‘poly’, ‘rbf’, ‘sigmoid’, ‘precomputed’ . Nó được sử dụng để tính toán trước ma trận hạt nhân từ ma trận dữ liệu; ma trận đó phải là một ma trận vuông (nxn). 

degree Mức độ của hàm nhân đa thức (‘poly’). Bỏ qua bởi tất cả các hạt nhân khác.

gamma: Hệ số hạt nhân cho ‘rbf’, ‘poly’ và ‘sigmoid’. Nếu gamma là ‘tự động’ thì 1 / n sẽ được sử dụng thay thế.

coef0: Thuật ngữ độc lập trong hàm hạt nhân. Nó chỉ có ý nghĩa trong ‘poly’ và ‘sigmoid’.

probability(xác suất): Cho dù để cho phép ước tính xác suất. Điều này phải được kích hoạt trước khi gọi phù hợp, và sẽ làm chậm phương pháp đó.

Shriking(thu hẹp):  Chọn dùng hay không dùng heuristic co rút.

tol: Dung sai cho tiêu chí dừng (kiểu try -catch mà sai nó vẫn cho chạy tiếp lol)

cache_size:Chỉ định kích thước của bộ đệm hạt nhân (tính bằng MB).

class_weight: Đặt tham số C của lớp i thành class_weight [i] * C cho SVC. Nếu không được, tất cả các lớp được cho là có trọng lượng một. Chế độ "cân bằng" sử dụng các giá trị của y để tự động điều chỉnh trọng số tỷ lệ nghịch với tần số lớp trong dữ liệu đầu vào là n_samples / (n_classes * np.bincount (y))

verbose:Cho phép báo cáo dài . Lưu ý rằng cài đặt này tận dụng lợi thế của cài đặt thời gian chạy trên mỗi quá trình trong libsvm, nếu được bật, có thể không hoạt động đúng trong ngữ cảnh đa luồng.

max_iter: Giới hạn cứng trên các lần lặp trong bộ giải, hoặc -1 cho không giới hạn.

decision_function_shape: ‘ovo’, ‘ovr’, mặc định = ’ovr’. Trả về hàm quyết định một lần so với (còn gọi là 'ovr') của hình dạng (n_samples, n_classes) như tất cả các bộ phân loại khác hay hàm quyết định ban đầu so với một ('ovo') của libsvm có hình dạng (n_samples) , n_classes * (n_classes - 1) / 2).

random_state: Hạt giống của trình tạo số ngẫu nhiên giả để sử dụng khi xáo trộn dữ liệu. Nếu int, random_state là hạt giống được sử dụng bởi trình tạo số ngẫu nhiên; Nếu RandomState instance, random_state là trình tạo số ngẫu nhiên; Nếu không, trình tạo số ngẫu nhiên là thể hiện RandomState được sử dụng bởi np.random.
